src/Microsoft.AspNetCore.Razor.Design/Internal/AssemblyTagHelperDescriptorResolver.cs

```diff
@@ -118,5 +126,103 @@ private void EnsureMvcInitialized()
             _populateMethodDelegate = (Action<IServiceCollection, string>) populateMethod
                 ?.CreateDelegate(typeof(Action<IServiceCollection, string>));
         }
+
+        private IEnumerable<TagHelperDescriptor> GetFakeDescriptors(IEnumerable<TagHelperDescriptor> vcthDescriptors)
+        {
+            var fakeDescriptors = new List<TagHelperDescriptor>();
+            foreach (var descriptor in vcthDescriptors)
+            {
+                var fakeType = typeof(ViewComponentTagHelperDesignTimeType).FullName;
+                var fakeDescriptor = new TagHelperDescriptor(descriptor);
+                fakeDescriptor.TypeName = fakeType;
+
+                var fakeAttributes = new List<TagHelperAttributeDescriptor>();
+                fakeDescriptor.Attributes = fakeAttributes;
+
+                var fakeHelperType = typeof(ViewComponentTagHelperDesignTimeHelperType).FullName;
+                var fakeHelperDescriptor = new TagHelperDescriptor(descriptor);
+                fakeHelperDescriptor.TypeName = fakeHelperType;
+
+                var fakeHelperAttributes = new List<TagHelperAttributeDescriptor>();
+                fakeHelperDescriptor.Attributes = fakeHelperAttributes;
+
+                foreach (var attribute in descriptor.Attributes)
+                {
+                    var summary = $"{attribute.TypeName}: {descriptor.TypeName}.{attribute.PropertyName}";
+                    var fakeVariableName = $"__obj";
+                    var fakeAttributePropertyName = $"{nameof(ViewComponentTagHelperDesignTimeType.PlaceholderProperty)} = null; " +
```

Comment:
Might want to individually comment on each piece of the injection for our own sake. Aka, we want to avoid "variable X" is not used so we do XY and Z